packages/upload-api/src/blob/accept.js

```diff
@@ -0,0 +1,62 @@
+import * as Server from '@ucanto/server'
+import * as W3sBlob from '@web3-storage/capabilities/web3.storage/blob'
+import { Assert } from '@web3-storage/content-claims/capability'
+import { create as createLink } from 'multiformats/link'
+import { Digest } from 'multiformats/hashes/digest'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { CAR } from '@ucanto/core'
+import * as API from '../types.js'
+import { BlobItemNotFound } from './lib.js'
+
+/**
+ * @param {API.W3ServiceContext} context
+ * @returns {API.ServiceMethod<API.BlobAccept, API.BlobAcceptSuccess, API.BlobAcceptFailure>}
+ */
+export function blobAcceptProvider(context) {
+  return Server.provideAdvanced({
+    capability: W3sBlob.accept,
+    handler: async ({ capability }) => {
+      const { blob } = capability.nb
+      // If blob is not stored, we must fail
+      const hasBlob = await context.blobsStorage.has(blob.content)
+      if (hasBlob.error) {
+        return {
+          error: new BlobItemNotFound(),
+        }
+      }
+
+      const digest = new Digest(sha256.code, 32, blob.content, blob.content)
+      const content = createLink(CAR.code, digest)
+      const w3link = `https://w3s.link/ipfs/${content.toString()}`
+
+      // TODO: Set bucket name
+      // TODO: return content commitment
+      const locationClaim = await Assert.location
+        .invoke({
+          issuer: context.id,
+          // TODO: we need user agent DID
```

Comment:
@Gozala I realized now that we do not have a good way to track this... Probably `accept` should also have `cause` CID like `blob/allocate` so that we can know this info.

Any ideas?

Reply:
Yeah I don't mind sticking cause here, but we could also add `space` field like we have in `allocate`

Reply:
I believe `accept` already has dependency on `allocate` via `ucan/await` syntax it just prefixed `_` implying that there is no need to pass an argument to a function. If we just want to capture link in the structure we probably don't need to do anything beyond that. However if we do want link to be passed into arguments then I do agree that we should add `cause`(al) link as you have suggested.